### PR TITLE
Issue #7: put more documentation in README.md

### PR DESCRIPTION
**Pull request by soxofaan**
Issue #7: put more documentation in README.md

### CODE REVIEW

**Comment by soxofaan** [comment]
rendered version of README.md can be found at https://github.com/soxofaan/pytest-unordered/blob/issue7-documentation/README.md at the moment

File: README.md [hunk 1]
```diff
@@ -1,18 +1,94 @@
...
+
+      E         Type mismatch:
+      E         <class 'tuple'> != <class 'list'>
+
+
+
+### Nesting
+
+A seasoned developer will notice that the simple use cases above
+can be easily be addressed with appropriate usage
```
Comment (utapyngo) [inline README.md:50]:
```suggestion
can easily be addressed with appropriate usage
```

**Review by utapyngo — COMMENTED** [review]
Awesome examples, thank you!
The only thing I want to add is a note about the `check_type` parameter at the end of the last section.

File: README.md [hunk 2]
```diff
@@ -1,18 +1,94 @@
...
+of builtins like `set`, `sorted` and `isinstance`.
```
Comment (utapyngo) [inline README.md:51]: `set` and `sorted` do not work with dictionaries.
`set([{'a': 2}, {'a': 1}]) == {{'a': 1}, {'a': 2}}` gives a TypeError, while `unordered([{'a': 2}, {'a': 1}]) == [{'a': 1}, {'a': 2}]` return True.

Comment (soxofaan) [reply README.md:51]: yes, indeed, good point. I'll try to include that use case

**Pull request merged**